CMakeLists.txt

```diff
@@ -43,14 +43,21 @@ if(WIN32 AND BUILD_SHARED_LIBS)
   set(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
 endif()
 
+# set the cxx standard version, and its extensions
+set(CMAKE_CXX_STANDARD 14 CACHE STRING "The C++ version to be used.")
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+message(STATUS "C++ Standard version: ${CMAKE_CXX_STANDARD}")
+
 if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18.0")
-  set(CMAKE_CXX_STANDARD 11 CACHE STRING "The C++ version to be used.")
+  set(CUDAToolkit_NVCC_EXECUTABLE "" CACHE STRING "The cudatoolkit nvcc path.")
   set(CUDAToolkit_LIBRARY_DIR "" CACHE STRING "The cudatoolkit library directory.")
   set(CUDAToolkit_INCLUDE_DIRS "" CACHE STRING "The cudatoolkit include directory.")
 
   find_package(CUDAToolkit REQUIRED)
   if(CUDAToolkit_FOUND)
     message(STATUS "found CUDAToolkit " ${CUDAToolkit_LIBRARY_DIR})
```

Comment:
Please separate the CMake changes into a separate commit, there are some things we'd need to discuss.

Reply:
it is OK if the others agree with these changes, but IIRC we made the standard 11 for a reason.

Reply:
11 is not remove, it's upgraded to 14, in upper of this code. As cub require 14.

Reply:
> but IIRC we made the standard 11 for a reason.

the reason is explained in the following comments:

https://github.com/danpovey/k2/pull/75#issuecomment-666194168
https://github.com/danpovey/k2/pull/75#issuecomment-666202006

> @csukuangfj why did you upgrade the build to c++14 from c++11, and what file did you edit to do that?
It's causing me problems building it, as the version of gcc here doesn't support that. I don't like to use any newer
version of c++ than we need to.

> OK thanks... yeah suppress the warning for now.  I guess it's more
efficient with c++14, but I'd like to make sure our code works with c++11,
so we can support a wider range of compilers.